lib/confluence.js

```diff
@@ -336,4 +336,77 @@ Confluence.prototype.updateAttachmentData = function(space, id, attachmentId, fi
         });
 };
 
+/**
+ *  This allows you to search by query
+ *
+ * @param {string} query
+ * @param {Function} callback
+ */
+Confluence.prototype.search = function(query, callback){
+
+    request
+        .get(this.config.baseUrl + this.config.apiPath + "/search" + this.config.extension + "?" + query)
+        .auth(this.config.username, this.config.password)
+        .end(function(err, res){
+            processCallback(callback, err, res);
+        });
+
+};
+
+/**
+ * Put/update data in labels a page.
+ *
+ * @param {string} space
+ * @param {string} id
+ * @param {number} version
+ * @param {string} title
+ * @param {array} content
+ * @param {Function} callback
+ */
+Confluence.prototype.putLabel = function(space, id, version, title, content, callback){
```

Comment:
I think we should revise this implementation to use confluence's [content/{id}/label](https://docs.atlassian.com/atlassian-confluence/REST/latest/#content/{id}/label) endpoint for this. Do you agree? If so, we can add the other verbs at some point.